src/scene/graph-node.js

```diff
@@ -597,39 +591,16 @@ class GraphNode extends EventHandler {
      * const node = parent.findOne('name', 'Test');
      */
     findOne(attr, value) {
+        const test = createTest(attr, value);
         const len = this._children.length;
-        let result = null;
 
-        if (attr instanceof Function) {
-            const fn = attr;
+        if (test(this))
+            return this;
 
-            result = fn(this);
+        for (let i = 0; i < len; ++i) {
+            const result = this._children[i].findOne(test);
```

Comment:
why not recurse on a function that directly accepts the test? this way we don't have to keep checking if it's a function at each and every invocation?